src/modules/fossil_metrics.rs

```diff
@@ -0,0 +1,267 @@
+use super::{Context, Module, ModuleConfig};
+
+use crate::configs::fossil_metrics::FossilMetricsConfig;
+use crate::formatter::StringFormatter;
+
+/// Creates a module with currently added/deleted lines in the Fossil check-out in the current
+/// directory.
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let mut module = context.new_module("fossil_metrics");
+    let config = FossilMetricsConfig::try_load(module.config);
+
+    // As we default to disabled=true, we have to check here after loading our config module,
+    // before it was only checking against whatever is in the config starship.toml
+    if config.disabled {
+        return None;
+    };
+
+    let is_checkout = context
+        .try_begin_scan()?
+        .set_files(&[".fslckout"])
+        .is_match();
+
+    if !is_checkout {
+        return None;
+    }
+
+    // Read the total number of added and deleted lines from "fossil diff --numstat"
+    let output = context.exec_cmd("fossil", &["diff", "--numstat"])?.stdout;
+    let stats = FossilDiff::try_parse(&output, config.only_nonzero_diffs)?;
+
+    let parsed = StringFormatter::new(config.format).and_then(|formatter| {
+        formatter
+            .map_style(|variable| match variable {
+                "added_style" => Some(Ok(config.added_style)),
+                "deleted_style" => Some(Ok(config.deleted_style)),
+                _ => None,
+            })
+            .map(|variable| match variable {
+                "added" => Some(Ok(stats.added)),
+                "deleted" => Some(Ok(stats.deleted)),
+                _ => None,
+            })
+            .parse(None, Some(context))
+    });
+
+    module.set_segments(match parsed {
+        Ok(segments) => segments,
+        Err(error) => {
+            log::warn!("Error in module `fossil_metrics`:\n{}", error);
+            return None;
+        }
+    });
+
+    Some(module)
+}
+
+/// Represents the parsed output from a Fossil diff with the --numstat option enabled.
+#[derive(Debug, PartialEq)]
+struct FossilDiff<'a> {
+    added: &'a str,
+    deleted: &'a str,
+}
+
+impl<'a> FossilDiff<'a> {
+    fn try_parse(diff_numstat: &'a str, only_nonzero_diffs: bool) -> Option<Self> {
+        let mut split = match diff_numstat.lines().last() {
+            Some(line) => line.split_whitespace(),
+            None => {
+                log::warn!("Error in module `fossil_metrics`:\nLast line of numstat diff missing.");
+                return None;
```

Comment:
``Error in module `fossil_metrics`:\n`` is a bit redundant, and the duplicate log message detector may have trouble with this. I would prefer having `try_parse` return a `Result<Self, &str>` instead. Please also add tests for some of the different error conditions.

Reply:
I didn't mean to say that you should be removing all error handling in the module, I think that is nice to have. The ``Error in module `fossil_metrics`:\n``-prefix was redundant, because the log message will already contain `WARN` and the module name.

Reply:
Okay, I see. I tried to move closer to the implementation of `git_metrics` where an unexpected output leads to 0 changes reported, e.g. if the format of the command output changes in an update or the `git` binary is not actually Git. The different error conditions explicitly checked for previously should not be able to occur if a Fossil version later than v2.3 (2017-07-21) is installed (which introduced the `--numstat` option). I can revert the latest commit and fix the log messages if preferable, but I'm uncertain about what other input to test when it cannot be produced by Fossil.